core/controllers/admin.py

```diff
@@ -1008,3 +1009,68 @@ def put(self):
         user_services.log_username_change(
             self.user_id, old_username, new_username)
         self.render_json({})
+
+
+class NumberOfDeletionRequestsHandler(base.BaseHandler):
+    """Handler for deleting account via admin page."""
+
+    GET_HANDLER_ERROR_RETURN_TYPE = feconf.HANDLER_TYPE_JSON
+
+    @acl_decorators.can_access_admin_page
+    def get(self):
+        self.render_json({
+            'number_of_pending_deletion_models': (
+                wipeout_service.get_number_of_pending_deletion_requests())
+        })
+
+
+class DeleteAccountHandler(base.BaseHandler):
+    """Handler for deleting account via admin page."""
+
+    @acl_decorators.can_access_admin_page
+    def post(self):
+        email = self.payload.get('email', None)
+
+        if email is None:
+            raise self.InvalidInputException(
+                'Invalid request: email must be specified')
+
+        if not isinstance(email, python_utils.UNICODE):
+            raise self.InvalidInputException(
+                'Expected email to be a unicode string, received %s' % email)
+
+        pending_deletion_request = (
+            wipeout_service.get_pending_deletion_request_by_email(email))
+        if pending_deletion_request is None:
+            raise self.InvalidInputException(
+                'There is no pending deletion request for email: %s' % email)
+
+        result = wipeout_service.run_user_deletion(pending_deletion_request)
+
+        self.render_json({'result': result})
+
+
+class VerifyAccountDeletedHandler(base.BaseHandler):
+    """Handler for verifying that account was deleted via admin page."""
+
+    @acl_decorators.can_access_admin_page
+    def post(self):
+        email = self.payload.get('email', None)
```

Comment:
Ditto, would have expected this to be done by username.

Also, don't we clean up the pending deletion requests? If so, then after that happens, is there no way to verify the deletion of an account? Under what circumstances would/should we run this function?

Reply:
We clean up the deletion request after we verify that the deletion was successful.

There will be a way to verify it since in #10762 I'm introducing the `DeletedUserModel`. This function is run after the deletion is completed to verify that the user was deleted properly.

Reply:
Done, removed.